### PR TITLE
Add missing NVIDIA_DEVICE_PLUGIN_FEATURE_ENABLED

### PR DESCRIPTION
**Pull request by arnaldo2792**
*Issue #, if available:*
N / A

*Description of changes:*
A change to include the NVIDIA Settings API for kubernetes was merged, but this constant is needed by downstreams to prevent accidental builds of the API.

*Testing*:
I used my own Settings SDK, and confirmed that I had access to the constant

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by vigh-m — APPROVED** [review]

**Review by cbgbt — APPROVED** [review]

**Review by sam-berning — APPROVED** [review]

**Comment by arnaldo2792** [comment]
TL;DR: closing in favor of a top level `settings.kubelet-device-plugin` settings extension, rather than an `nvidia-device-plugin` cargo feature.

In the current implementation, we favored `nvidia-device-plugin` as a cargo feature so that downstream builders can extend their API definitions by enabling cargo features that extend the API model. However, we pivoted towards having a top level API along with its settings extension because we have use cases where the API shape conflicts with what the program expects. One example of this is `pluto` in the Bottlerocket Core Kit, it would have to keep enabling all the features added to the top level API for the deserialization to succeed, even though the program only cares about top level Kubernetes settings. 

To keep programs independent from whatever the downstream decide to consume in their API, we decided that it will be best to provide top level APIs instead of extending structs with cargo features.
